internal/databases/mysql/binlog_server_handler.go

```diff
@@ -0,0 +1,239 @@
+package mysql
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"net"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-mysql-org/go-mysql/mysql"
+	"github.com/go-mysql-org/go-mysql/replication"
+	"github.com/go-mysql-org/go-mysql/server"
+	"github.com/google/uuid"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+var (
+	untilTS      time.Time
+	lastSentGTID string
+)
+
+func addRotateEvent(s *replication.BinlogStreamer, pos mysql.Position) error {
+	// create rotate event
+	rotateBinlogEvent := replication.BinlogEvent{}
+	rotateBinlogEvent.RawData = make([]byte, 4+1+4+4+4+2+8)
```

Comment:
This magic number should be placed in const var with some description and links in comment : )